packages/viewer/src/components/scene-tree-table-layout/scene-tree-table-layout.tsx

```diff
@@ -258,6 +265,23 @@ export class SceneTreeTableLayout {
 
   public async componentWillRender(): Promise<void> {
     await this.computeAndUpdateViewportRows();
+
+    const isViewportDataPresent =
+      this.rows[this.viewportStartIndex] != null ||
+      this.rows[this.viewportEndIndex] != null;
+
+    if (!isViewportDataPresent && this.loadingTimer == null) {
+      this.loadingTimer = window.setTimeout(
+        this.setIsLoading,
+        DISPLAY_LOADER_TIMEOUT
+      );
+    } else if (isViewportDataPresent) {
+      this.isStillLoading = false;
+      if (this.loadingTimer) {
+        clearTimeout(this.loadingTimer);
+      }
+      this.loadingTimer = undefined;
+    }
```

Comment:
This is pretty fragile. We're using the rendering mechanics of Stencil to infer loading status..

What if we added a `loading` field to `SceneTreeState` and have the `SceneTreeController` manage this field. This way we can bind the view to the current state.